Library/Homebrew/install.rb

```diff
@@ -300,11 +301,14 @@ def install_formulae(
           debug:                      debug,
           quiet:                      quiet,
           verbose:                    verbose,
+          dry_run:                    dry_run,
         )
 
         begin
-          fi.prelude
-          fi.fetch
+          unless dry_run
+            fi.prelude
+            fi.fetch
+          end
```

Comment:
Given `formula_installer` gets a `dry_run` argument it feels like this should be handled in there. I'm not sure I understand, though, why we need the `formula_installer` at all for a `dry_run` considering we're doing messaging in this class?

Reply:
Could the `dry_run` logic happen before here so the `unless dry_run` is no longer needed?